utils/build/docker/java/spring-boot/src/main/java/com/datadoghq/system_tests/springboot/App.java

```diff
@@ -438,6 +448,47 @@ RedirectView traceRedirect(@RequestParam(required = false, name="url") String re
         return new RedirectView("https://" + redirect);
     }
 
+    @RequestMapping("/e2e_otel_span")
+    String e2eOtelSpan(@RequestHeader(name = "User-Agent") String userAgent,
+                       @RequestParam(name="parentName") String parentName,
+                       @RequestParam(name="childName") String childName,
+                       @RequestParam(required = false, name="shouldIndex") int shouldIndex) {
+        // Get an OpenTelemetry tracer
+        Tracer tracer = GlobalOpenTelemetry.getTracer("system-tests");
+        // Aggregate common attributes
+        AttributesBuilder commonAttributesBuilder = Attributes.builder()
+                .put("http.useragent", userAgent);
+        if (shouldIndex == 1) {
+            commonAttributesBuilder.put("_dd.filter.kept", 1);
+            commonAttributesBuilder.put("_dd.filter.id", "system_tests_e2e");
+        }
+        Attributes commonAttributes = commonAttributesBuilder.build();
+        // Create parent span according test requirements
+        SpanBuilder parentSpanBuilder = tracer.spanBuilder(parentName)
+                .setNoParent()
+                .setAllAttributes(commonAttributes)
+                .setAttribute("attributes", "values");
+        io.opentelemetry.api.trace.Span parentSpan = parentSpanBuilder.startSpan();
+        parentSpan.setStatus(ERROR, "testing_end_span_options");
+
+        try (Scope scope = parentSpan.makeCurrent()) {
+            // Create child span according test requirements
+            Instant now = Instant.now();
```

Comment:
Shouldn’t `now` be used as the start time for the `parent` span?

Reply:
It is not part of the test requirements. 
I could but I feel it will be more confusing than anything 🤷

Reply:
Ah, I misread the code as well. Both `now` and `oneSecLater` is used on the child span. Then they make some sense.

Reply:
Yeah, I tried to put the timing under the `// Create child span according test requirements` comment umbrella.
Is everything okay for you then?